aclk/aclk_query.c

```diff
@@ -0,0 +1,574 @@
+#include "aclk_common.h"
+#include "aclk_query.h"
+#include "aclk_stats.h"
+
+pthread_cond_t query_cond_wait = PTHREAD_COND_INITIALIZER;
+pthread_mutex_t query_lock_wait = PTHREAD_MUTEX_INITIALIZER;
+#define QUERY_THREAD_LOCK pthread_mutex_lock(&query_lock_wait)
+#define QUERY_THREAD_UNLOCK pthread_mutex_unlock(&query_lock_wait)
+
+volatile int aclk_connected = 0;
+
+#ifndef __GNUC__
+#pragma region ACLK_QUEUE
```

Comment:
I think this will break most non-gcc compilers such as clang.

Reply:
Clang has support as far as I know

Reply:
1. Microsoft Documentation on the feature:
https://docs.microsoft.com/en-us/cpp/preprocessor/region-endregion

1. LLVM change which enabled `pragma region` across the board:
https://reviews.llvm.org/D42248

1. GCC change request https://gcc.gnu.org/bugzilla//show_bug.cgi?id=85487

Reply:
FreeBSD 12 does not complain so I guess we are ok.